pyvista/core/filters.py

```diff
@@ -1841,118 +1841,188 @@ def interpolate(dataset, target, sharpness=2, radius=1.0,
-        those associated with streamline-points are stored in the point data.
-
-        This uses a Sphere as the source - set it's location and radius via
-        the ``source_center`` and ``source_radius`` keyword arguments.
-        You can retrieve the source as :class:`pyvista.PolyData` by specifying
-        ``return_source=True``.
+        The default behavior uses a Sphere as the source - set it's location and 
+        radius via the ``source_center`` and ``source_radius`` keyword arguments.
+        ``n_points`` defines the number of starting points on the sphere surface.
+        Alternatively, a Line source can be used by specifying ``pointa`` and ``pointb``.
+        ``n_points`` again defines the number of points on the line.
+        
+        You can retrieve the source by specifying ``return_source=True``.
+        
+        Optional keyword parameters from ``streamlines_from_source`` can be used here to
+        control the generation of streamlines.
 
         Parameters
         ----------
-        vectors : str
-            The string name of the active vector field to integrate across
+        vectors : str, optional
+            The string name of the active vector field to integrate across.
 
-        source_center : tuple(float)
+        source_center : tuple(float), optional
             Length 3 tuple of floats defining the center of the source
-            particles. Defaults to the center of the dataset
+            particles. Defaults to the center of the dataset.
 
-        source_radius : float
+        source_radius : float, optional
             Float radius of the source particle cloud. Defaults to one-tenth of
-            the diagonal of the dataset's spatial extent
+            the diagonal of the dataset's spatial extent.
+
+        n_points : int, optional
+            Number of particles present in source sphere or line.
+
+        start_position : tuple(float), optional
+            A single point.  This will override the sphere point source.
+
+        return_source : bool, optional
+            Return the source particles as :class:`pyvista.PolyData` as well as the
+            streamlines. This will be the second value returned if ``True``.
+
+        pointa, pointb : tuple(float), optional
+            The coordinates of a start and end point for a line source. This
+            will override the sphere and start_position point source.
+
+        Returns:
+        --------
+        streamlines : pyvista.PolyData
+            This produces polylines as the output, with each cell
+            (i.e., polyline) representing a streamline. The attribute values
+            associated with each streamline are stored in the cell data, whereas
+            those associated with streamline-points are stored in the point data.
+
+        source : pyvista.PolyData
+            The points of the source are the seed points for the streamlines.
+            Only returned if `return_source=True`.
```

Comment:
Restructured text style requires two `
```suggestion
            Only returned if ``return_source=True``.
```